components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/UserSessionDAO.java

```diff
@@ -22,11 +22,21 @@
         .SessionManagementServerException;
 import org.wso2.carbon.identity.application.authentication.framework.model.UserSession;
 
+import java.util.Map;
+
 /**
  * Perform operations for {@link UserSession}.
  */
 public interface UserSessionDAO {
 
     UserSession getSession(String sessionId) throws SessionManagementServerException;
 
+    /**
+     * Get federated authentication session mapping info during the federated idp init logout
+     *
+     * @param fedIdpSessionId
+     * @return A map containing federated authentication session details
+     * @throws SessionManagementServerException
+     */
+    Map<String, String> getSessionDetails(String fedIdpSessionId) throws SessionManagementServerException;
```

Comment:
Shall we rename the method to reflect that the session retrieved is for a federated user?

Reply:
Renamed in 845675d